crates/ruff_server/src/server/api/requests/execute_command.rs

```diff
@@ -43,16 +51,14 @@ impl super::SyncRequestHandler for ExecuteCommand {
             return Err(anyhow::anyhow!("Cannot execute the '{}' command: the client does not support `workspace/applyEdit`", command.label())).with_failure_code(ErrorCode::InternalError);
         }
 
-        let mut changes = HashMap::new();
+        let mut edit_tracker =
+            EditTracker::new(session.resolved_client_capabilities().document_changes);
```

Comment:
I would suggest perhaps passing in whatever's returned by `session.resolved_client_capabilities()` to reduce the risk of passing in a wrong boolean.